program_intent_engagement/apps/api/v1/tests/test_views.py

```diff
@@ -0,0 +1,139 @@
+"""
+Tests for the program intent API views
+"""
+import json
+import uuid
+
+import ddt
+from django.urls import reverse
+
+from program_intent_engagement.apps.core.models import ProgramIntent
+from test_utils import ProgramIntentsAPITestCase
+
+
+@ddt.ddt
+class ProgramIntentsViewTests(ProgramIntentsAPITestCase):
+    """
+    Tests Program Intents View
+    """
+
+    def setUp(self):
+        super().setUp()
+
+        self.url = reverse('api:v1:program_intent-insert')
+
+        self.program_uuid = str(uuid.uuid4())
+        self.data = {
+            'program_uuid': self.program_uuid,
+            'reason': 'ALL_COURSES_COMPLETE',
+            'certainty': 'CERTAIN_YES',
+            'effective_timestamp': '2022-07-01 00:00:00',
+        }
+
+    def post_api(self, user, data):
+        """
+        Helper function to make a POST request to the API
+        """
+        data = json.dumps(data)
+        headers = self.build_jwt_headers(user)
+
+        return self.client.post(self.url, data, **headers, content_type="application/json")
+
+    def get_response(self, user, data, expected_response):
+        """
+        Helper function to get API response
+        """
+        response = self.post_api(user, data)
+        self.assertEqual(response.status_code, expected_response)
+
+        return response
+
+    # def test_auth_failures(self):
+    #     """
+    #     Test that the endpoint correctly validates permissions.
+    #     """
+    #     response = self.client.post(self.url)
+    #     self.assertEqual(response.status_code, 401)
+
+    # todo: is this actually the expected behavior?
+    def test_empty_program_intent(self):
+        """
+        Test that the endpoint returns 400 if input is empty
+        """
+        self.get_response(self.user, {}, 400)
+
+    def test_valid_program_intent(self):
+        """
+        Test that the endpoint returns 200 if input is valid
+        """
+        self.get_response(self.user, self.data, 200)
+        self.assertEqual(len(ProgramIntent.objects.all()), 1)
+        intent = ProgramIntent.objects.get(program_uuid=self.program_uuid,
```

Comment:
naming nit - you expect this return to be not an intent so I'd just call it response

Reply:
no you're completely right, I just can't read, your assert is NOT none
I read it as assertNone and then my brain changed all the code before it to match
these names are all correct

Reply:
No worries - gotta love a double negative :P 

Thanks for verifying that.